modules/statistics/RTPStatsCollector.js

```diff
@@ -940,3 +967,342 @@ StatsCollector.prototype.processAudioLevelReport = function() {
 };
 
 /* eslint-enable no-continue */
+
+/**
+ * New promised based getStats report processing.
+ * Tested with chrome, firefox and safari. Not switching it on for for chrome as
+ * frameRate stat is missing and calculating it using framesSent,
+ * gives values double the values seen in webrtc-internals.
+ * https://w3c.github.io/webrtc-stats/
+ */
+
+/**
+ * Defines a function which (1) is to be used as a StatsCollector method and (2)
+ * gets the value from a specific report returned by RTCPeerConnection#getStats
+ * associated with a lib-jitsi-meet browser-agnostic name in case of using
+ * Promised based getStats.
+ *
+ * @param {Object.<string,string>} keys the map of LibJitsi browser-agnostic
+ * names to RTCPeerConnection#getStats browser-specific keys
+ */
+StatsCollector.prototype._defineNewGetStatValueMethod = function(keys) {
+    // Define the function which converts a lib-jitsi-meet browser-asnostic name
+    // to a browser-specific key of a report returned by
+    // RTCPeerConnection#getStats.
+    const keyFromName = function(name) {
+        const key = keys[name];
+
+        if (key) {
+            return key;
+        }
+
+        // eslint-disable-next-line no-throw-literal
+        throw `The property '${name}' isn't supported!`;
+    };
+
+    // Compose the 2 functions defined above to get a function which retrieves
+    // the value from a specific report returned by RTCPeerConnection#getStats
+    // associated with a specific lib-jitsi-meet browser-agnostic name.
+    return (item, name) => item[keyFromName(name)];
+};
+
+/**
+ * Converts the value to a non-negative number.
+ * If the value is either invalid or negative then 0 will be returned.
+ * @param {*} v
+ * @return {number}
+ * @private
+ */
+StatsCollector.prototype.getNonNegativeValue = function(v) {
```

Comment:
We use this method to get the ssrc, the bytes, the packets and the lost packets (negative because of retransmissions, maybe?). These look like unsigned positive ints, so I'm a bit confused why we need this (I'm not debating whether we need it or not, I'm just trying to understand why we need it). So, are we flipping the sign in an effort to work around some webrtc bug? If so, can we add a small comment documenting this? If there's a bug report on the webrtc side of things, we can include that too.